### PR TITLE
Ingestigate GG Matplotlib Functions

### LINKED ISSUE #1602: [GraphGenerator] Investigate Current Graph Options in GG
There are around 30 different graph options listed in the `MATPLOTLIB_PLOT_FUNCTIONS` dict in GG. Some of these functions need 2 sets of data to work properly and some only need 1. We know `scatter` and `stackplot` are examples of tuple based graphs and `plot` is an example of a graph type that only needs one set of data.

## Expected Behavior

Any and every graph function in the `MATPLOTLIB_PLOT_FUNCTIONS` dict in GG should be usable.

## Current Behavior

It was discovered recently in PR #1594 that `scatter` was not functioning properly but it immediately started working when it was added to the `TUPLE_BASED_FUNCTIONS` list in GG. Once this is merged in, we will have 3 graph types that are for sure working: `plot`, `stackplot`, and `scatter`.

## Possible Solution

The remaining graph types should be checked for functionality and if they're not working right now, it should be checked if they are actually tuple-based by adding the string-key from the `MATPLOTLIB_PLOT_FUNCTIONS` dict to the `TUPLE_BASED_FUNCTIONS` list.

If they don't work, make a note of it and we can determine if the functionality is necessary for further investigation.

## Steps to Reproduce

1. Add any graph type outside of `scatter`, `stackplot`, and `plot` (assuming PR #1594 has been merged in for `scatter` to function properly) as the `type` in a graph filter file.
2. Run a simulation and see if the graph type is producing properly.

### PR DESCRIPTION
**Pull request by ew3361zh**
Check which matplotlib options in GG are currently functional and try to ensure non-functional graphing options are noted and logged to the user if they attempt to use them.

### Context

Issue(s) closed by this pull request: closes #1602

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.
      
How big are the changes in the PR? Would you nominate it for a major version upgrade?
- [ ] Yes
- [x] No

### What

Check and document the functionality of each of the GG matplotlib options (i.e. everything in the `MATPLOTLIB_PLOT_FUNCTIONS` global dict in `graph_generator.py`) as both a non-Tuple-base and Tuple-based function.

Enable functionality of the Tuple-based functions that were not previously known to be Tuple-based `("barbs", "hexbin", "quiver", "spy")`.

Log unsupported graph types to OM to let user know why their graph didn't work `("area", "bar", "broken_barh", "contour", "filled-contour", "hist2d",
                                          "horizontal_bar", "horizontal_line", "horizontal_lines", "imshow",
                                          "pcolor", "pcolormesh", "quiver_key", "step", "streamplot", "tripcolor",
                                          "vertical_line", "vertical_lines")`

### Why

Not all options in the GG were supported and someone unsuspectingly using an unsupported option could get an unexpected and unhelpful error message.
Additionally, some of the options that were unsupported in their current state were in fact supported as Tuple-based functions so adding those to the list of `TUPLE_BASED_FUNCTIONS` enables their functionality.

### How

1. Add all Tuple-based options to `TUPLE_BASED_FUNCTIONS` global list.
2. Add all unsupported options to `UNSUPPORTED_GRAPH_FUNCTIONS` list and add a check and log an error to the user if they select one of these options.
3. Add comment linking research of what the result was for each function:
[https://docs.google.com/spreadsheets/d/10fPdoS5YejYPidYvAmEBkMNq0X9nMqYdta-qduOk42s/edit#gid=0](url)

### Test plan

Tried plotting some graphs using the newly enabled Tuple-based options.
Tried plotting some graphs using the newly tracked unsupported options with verbosity set to `errors` to see the errors as expected.

### CODE REVIEW

File: tests/test_graph_generator.py [hunk 1]
```diff
@@ -78,23 +78,52 @@ def test_generate_graph_path_no_title(graph_generator: GraphGenerator) -> None:
...
+        ),
+    ],
+)
+def test_generate_graph_without_producing_graphics(
+    graph_generator: GraphGenerator,
+    graph_details: list[dict[str, str]],
+    expected_output: dict[str, str | dict[str, Any]],
+    produce_graphics: bool,
+) -> None:
     filtered_pool = {"dummy_key": {"dummy_data": [1, 2, 3]}}
```
Comment (matthew7838) [inline tests/test_graph_generator.py]: Missing docstrings.

**Review by matthew7838 — APPROVED** [review]
LGTM, minor catch, missing docstrings.

File: tests/test_graph_generator.py [hunk 2]
```diff
@@ -78,23 +78,52 @@ def test_generate_graph_path_no_title(graph_generator: GraphGenerator) -> None:
...
-    graph_details = {"title": "Example Graph"}
+    graph_details = graph_details
```
Comment (Isgotkowitz) [inline tests/test_graph_generator.py]:
```suggestion
```
I think this isn't needed.

File: tests/test_graph_generator.py [hunk 3]
```diff
@@ -78,23 +78,52 @@ def test_generate_graph_path_no_title(graph_generator: GraphGenerator) -> None:
...
-            "message": "'produce_graphics' set to False, no graphs will be produced.",
-            "info_map": {
-                "class": graph_generator.__class__.__name__,
-                "function": "generate_graph",
-            },
-        }
-    ]
+    produce_graphics = produce_graphics
+
+    expected_output = expected_output
```
Comment (Isgotkowitz) [inline tests/test_graph_generator.py]:
```suggestion
```
I don't think these are needed either.

**Review by Isgotkowitz — APPROVED** [review]
LGTM. Prints errors appropriately, all the new tuple-based plotting types work as described in the google sheet.

**Pull request merged**